src/Halcyon/HAL/HALResponse.cs

```diff
@@ -26,9 +25,9 @@ public class HALResponse {
         public HALResponse(object model, IHALModelConfig config = null)
             : this(config) {
             if (!(model is JObject) && (model is IEnumerable)) {
-                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", "model");
+                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", nameof(model));
             }
-
+            HALAttributeResolver.ResolveAttributes(this, model);
```

Comment:
I think this is really important and it should work well.

I don't see much of a use case for passing in a decorated model into a `HALResponse`. Models that are decorated with Attributes should not need more links or embedded collections added to them. Even in the rare case that a user might need to add further links and embedded collections to models, they could always use the `IHALConverter` directly.

Attribute models will be passed into `this.Ok(model)` in the vast majority of cases. The serializer will then pick up it is an attribute model and convert it.

This allows for a better separation of concerns and testability. Let me know how you go.